utils/update.php

```diff
@@ -68,11 +82,13 @@
         echo "Does the URL point to a directory containing OSM update data?\n\n";
         fail('replication URL not reachable.');
     }
-    $sSetup = CONST_InstallPath.'/utils/setup.php';
-    $iRet = -1;
-    passthru($sSetup.' --create-functions --enable-diff-updates', $iRet);
-    if ($iRet != 0) {
-        fail('Error running setup script');
+    if ($aResult['update-functions']) {
+        $sSetup = CONST_InstallPath.'/utils/setup.php';
+        $iRet = -1;
+        passthru($argv[0].' '.$sSetup.' --create-functions --enable-diff-updates', $iRet);
```

Comment:
Wouldn't $argv[0] be the name of the executing script? I don't see how that belongs here.

Reply:
My intent was to account for the case where the `php` interpreter was executed from a location other than `/usr/bin/php` (as stated in the interpreter directive at the top of `setup.php`). In that case, running `setup.php` by itself would fail. `$argv[0]` should reference the interpreter executable path, but now that I'm looking at it again, I think you're right that PHP populates `$argv[0]` with the script path instead of the actual system `argv[0]`.

I can revert this, unless you have a suggestion for executing the same interpreter executable that is running the `update.php` script.

Reply:
PHP_BINARY would be the way to go but this is really something for a different PR. I suspect that there are other places that need to be adapted.

Reply:
Agreed that a separate PR makes the most sense. I'll take it out of this one.